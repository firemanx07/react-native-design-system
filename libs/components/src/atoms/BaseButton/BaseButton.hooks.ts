import { useNamespacedTheme } from '@proxym/themes';
import { rgba } from 'polished';

import { ButtonVariant } from './BaseButton.types';

export const useButtonContentColor = (
  variant: ButtonVariant,
  disabled?: boolean,
) => {
  const { colors } = useNamespacedTheme();

  switch (variant) {
    case ButtonVariant.Outline:
      return disabled ? rgba(colors.dark, 0.3) : colors.dark;
    case ButtonVariant.Primary:
    case ButtonVariant.Secondary:
    default:
      return colors.light;
  }
};
