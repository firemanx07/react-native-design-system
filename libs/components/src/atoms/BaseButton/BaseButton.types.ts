export enum ButtonVariant {
  Primary = 'Primary',
  Secondary = 'Secondary',
  Outline = 'Outline',
}

export enum ButtonSize {
  Regular = 'Regular',
  Medium = 'Medium',
  Small = 'Small',
}

export enum ButtonShape {
  Rounded = 'Rounded',
  Square = 'Square',
}
