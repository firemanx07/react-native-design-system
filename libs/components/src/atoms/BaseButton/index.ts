export { default as BaseButton } from './BaseButton';
export { ButtonVariant, ButtonSize, ButtonShape } from './BaseButton.types';
