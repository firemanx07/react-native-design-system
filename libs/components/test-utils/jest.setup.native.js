import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
