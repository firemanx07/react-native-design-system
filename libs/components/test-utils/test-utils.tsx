import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
