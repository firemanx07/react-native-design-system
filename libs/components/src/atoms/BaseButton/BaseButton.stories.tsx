import { ColorType } from '@proxym/themes';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import { StorybookScreen, StorybookSection } from '../../storybook';
import { PersonIcon } from '../Icons';
import {
  BaseButton as BaseButtonBase,
  PropsType as BaseButtonProps,
} from './BaseButton';
import { ButtonSize, ButtonVariant } from './BaseButton.types';

const Story: ComponentMeta<typeof BaseButtonBase> = {
  component: BaseButtonBase,
  title: 'Atoms/Base Button',
  argTypes: {
    customColor: {
      control: { type: 'select' },
      options: ColorType,
    },
    variant: {
      control: { type: 'radio' },
    },
    size: {
      control: { type: 'radio' },
    },
  },
};
export default Story;

const Template: ComponentStory<typeof BaseButtonBase> = (
  args: BaseButtonProps,
) => (
  <StorybookScreen>
    <StorybookSection>
      <BaseButtonBase {...args} />
    </StorybookSection>
  </StorybookScreen>
);

const renderIcon = (size: number, color: string) => (
  <PersonIcon width={size} height={size} fill={color} />
);

const parameters = {
  controls: {
    include: [
      'size',
      'customColor',
      'disabled',
      'loading',
      'children',
      'customColor',
    ],
  },
};

const defaultArgs: Omit<BaseButtonProps, 'variant' | 'renderIcon'> = {
  size: ButtonSize.Regular,
  disabled: false,
  loading: false,
  children: 'Button Text',
  onPress: () => {
    // This is a mock
  },
};

export const Primary = Template.bind({});
Primary.parameters = parameters;
Primary.args = {
  variant: ButtonVariant.Primary,
  ...defaultArgs,
};

export const PrimaryWithIcon = Template.bind({});
PrimaryWithIcon.parameters = parameters;
PrimaryWithIcon.args = {
  variant: ButtonVariant.Primary,
  renderIcon,
  ...defaultArgs,
};

export const Secondary = Template.bind({});
Secondary.parameters = parameters;
Secondary.args = {
  variant: ButtonVariant.Secondary,
  ...defaultArgs,
};

export const SecondaryWithIcon = Template.bind({});
SecondaryWithIcon.parameters = parameters;
SecondaryWithIcon.args = {
  variant: ButtonVariant.Secondary,
  renderIcon,
  ...defaultArgs,
};

export const Outline = Template.bind({});
Outline.parameters = parameters;
Outline.args = {
  variant: ButtonVariant.Outline,
  ...defaultArgs,
};

export const OutlineWithIcon = Template.bind({});
OutlineWithIcon.parameters = parameters;
OutlineWithIcon.args = {
  variant: ButtonVariant.Outline,
  renderIcon,
  ...defaultArgs,
};
