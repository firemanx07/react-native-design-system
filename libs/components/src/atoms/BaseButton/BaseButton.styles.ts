import { css, styled } from '@proxym/themes';
import ColorType from '@proxym/themes/lib/typescript/src/types/ColorType';
import { rgba } from 'polished';

import { BaseText, TextVariant, TextWeight } from '../BaseText';
import { ButtonShape, ButtonSize } from './BaseButton.types';

const buttonHeight = {
  [ButtonSize.Regular]: 48,
  [ButtonSize.Medium]: 36,
  [ButtonSize.Small]: 32,
};

export const iconSize = {
  [ButtonSize.Regular]: 24,
  [ButtonSize.Medium]: 20,
  [ButtonSize.Small]: 16,
};

const buttonTextVariant = {
  [ButtonSize.Regular]: TextVariant.L,
  [ButtonSize.Medium]: TextVariant.M,
  [ButtonSize.Small]: TextVariant.M,
};

const buttonStylesBySize = {
  [ButtonSize.Regular]: css`
    padding-horizontal: ${({ theme }) => theme.ds.spacing.medium}px;
  `,
  [ButtonSize.Medium]: css`
    min-width: 140px;
    padding-horizontal: ${({ theme }) => theme.ds.spacing.medium}px;
    align-self: flex-start;
  `,
  [ButtonSize.Small]: css`
    padding-horizontal: ${({ theme }) => theme.ds.spacing.primary}px;
    align-self: flex-start;
  `,
};

type ButtonType = {
  size: ButtonSize;
  shape: ButtonShape;
  disabled?: boolean;
  customColor?: string;
};

const ButtonCommon = styled.TouchableOpacity<ButtonType>`
  ${({ size }) => buttonStylesBySize[size]}
  height: ${({ size }) => buttonHeight[size]}px;
  border-radius: ${({ size, shape }) => {
    switch (shape) {
      case ButtonShape.Rounded:
        return buttonHeight[size] / 2;
      case ButtonShape.Square:
      default:
        return 7;
    }
  }}px;
  max-width: 100%;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;
const isDisabled = (
  colors: typeof ColorType,
  disabled: boolean | undefined,
  customColor: string | undefined,
) =>
  !disabled &&
  `border-color: ${customColor ?? colors.primary};
    border-width: 1px;`;
export const ButtonPrimary = styled(ButtonCommon)`
  background-color: ${({ theme, disabled, customColor }) =>
    disabled
      ? rgba(theme.ds.colors.grayDark, 0.2)
      : customColor ?? theme.ds.colors.primary};
`;

export const ButtonSecondary = styled(ButtonCommon)`
  background-color: ${({ theme, disabled, customColor }) =>
    disabled
      ? rgba(theme.ds.colors.grayDark, 0.2)
      : customColor ?? theme.ds.colors.secondary};
`;

export const ButtonOutline = styled(ButtonCommon)`
  background-color: ${({ theme, disabled }) =>
    disabled ? rgba(theme.ds.colors.grayDark, 0.2) : theme.ds.colors.light};
  ${({ theme, disabled, customColor }) =>
    isDisabled(theme.ds.colors, disabled, customColor)}
`;

type ButtonTextType = {
  size: ButtonSize;
  color: string;
  hasIcon?: boolean;
};

export const ButtonText = styled(BaseText).attrs(
  ({ size }: ButtonTextType) => ({
    variant: buttonTextVariant[size],
    weight: TextWeight.Bold,
  }),
)<ButtonTextType>`
  color: ${({ color }) => color};
  margin-left: ${({ hasIcon, size }) => {
    const iconSize = size === ButtonSize.Small ? 4 : 8;
    return hasIcon ? iconSize : 0;
  }}px;
`;
