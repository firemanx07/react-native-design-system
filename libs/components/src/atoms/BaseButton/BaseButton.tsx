import React, { memo, useMemo, ReactNode } from 'react';
import { StyleProp, TextStyle, ViewStyle, TextProps } from 'react-native';

import { TestIDType } from '../../types';
import { LoadingIndicator } from '../LoadingIndicator';
import { useButtonContentColor } from './BaseButton.hooks';
import {
  ButtonPrimary,
  ButtonSecondary,
  ButtonOutline,
  iconSize,
  ButtonText,
} from './BaseButton.styles';
import { ButtonSize, ButtonVariant, ButtonShape } from './BaseButton.types';

export type PropsType = {
  variant?: ButtonVariant;
  size?: ButtonSize;
  children: TextProps['children'];
  onPress?: () => void;
  loading?: boolean;
  disabled?: boolean;
  renderIcon?: (size: number, color: string) => ReactNode;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  shape?: ButtonShape;
} & TestIDType;

export const BaseButton = ({
  variant = ButtonVariant.Primary,
  size = ButtonSize.Regular,
  shape = ButtonShape.Rounded,
  onPress,
  children,
  disabled,
  loading,
  style,
  textStyle,
  renderIcon,
  testID,
}: PropsType) => {
  const contentColor = useButtonContentColor(variant, disabled);
  const hasIcon = !!renderIcon;

  const Button = useMemo(() => {
    switch (variant) {
      case ButtonVariant.Secondary:
        return ButtonSecondary;
      case ButtonVariant.Outline:
        return ButtonOutline;
      case ButtonVariant.Primary:
      default:
        return ButtonPrimary;
    }
  }, [variant]);

  const renderContent = () => {
    if (!disabled && loading) {
      return <LoadingIndicator size={8} color={contentColor} />;
    }

    return (
      <>
        {hasIcon && renderIcon(iconSize[size], contentColor)}
        <ButtonText
          size={size}
          color={contentColor}
          style={textStyle}
          numberOfLines={1}
          hasIcon={hasIcon}
        >
          {children}
        </ButtonText>
      </>
    );
  };

  return (
    <Button
      size={size}
      shape={shape}
      disabled={disabled}
      style={style}
      onPress={onPress}
      testID={testID}
    >
      {renderContent()}
    </Button>
  );
};

export default memo(BaseButton);
