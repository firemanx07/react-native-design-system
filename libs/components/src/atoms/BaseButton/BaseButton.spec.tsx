import { testAllStories } from '../../../test-utils';
import * as stories from './BaseButton.stories';

describe('BaseButton tests', () => {
  testAllStories(stories);
});
