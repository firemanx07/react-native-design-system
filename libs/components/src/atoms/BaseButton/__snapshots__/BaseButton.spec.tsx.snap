// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton tests Outline snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4D37D9",
          "borderRadius": 7,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <Text
        color="#4D37D9"
        fontFamily="Rubik-Bold"
        fontSize={16}
        hasIcon={false}
        lineHeight={20}
        numberOfLines={1}
        size="Regular"
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "color": "#4D37D9",
              "marginLeft": 0,
            },
          ]
        }
      >
        Button Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`BaseButton tests OutlineWithIcon snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4D37D9",
          "borderRadius": 7,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <test-file-stub
        fill="#4D37D9"
        height={24}
        width={24}
      />
      <Text
        color="#4D37D9"
        fontFamily="Rubik-Bold"
        fontSize={16}
        hasIcon={true}
        lineHeight={20}
        numberOfLines={1}
        size="Regular"
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "color": "#4D37D9",
              "marginLeft": 8,
            },
          ]
        }
      >
        Button Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`BaseButton tests Primary snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4D37D9",
          "borderRadius": 7,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <Text
        color="#FFFFFF"
        fontFamily="Rubik-Bold"
        fontSize={16}
        hasIcon={false}
        lineHeight={20}
        numberOfLines={1}
        size="Regular"
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "color": "#FFFFFF",
              "marginLeft": 0,
            },
          ]
        }
      >
        Button Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`BaseButton tests PrimaryWithIcon snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4D37D9",
          "borderRadius": 7,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <test-file-stub
        fill="#FFFFFF"
        height={24}
        width={24}
      />
      <Text
        color="#FFFFFF"
        fontFamily="Rubik-Bold"
        fontSize={16}
        hasIcon={true}
        lineHeight={20}
        numberOfLines={1}
        size="Regular"
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "color": "#FFFFFF",
              "marginLeft": 8,
            },
          ]
        }
      >
        Button Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`BaseButton tests Secondary snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#49C4E9",
          "borderRadius": 7,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <Text
        color="#FFFFFF"
        fontFamily="Rubik-Bold"
        fontSize={16}
        hasIcon={false}
        lineHeight={20}
        numberOfLines={1}
        size="Regular"
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "color": "#FFFFFF",
              "marginLeft": 0,
            },
          ]
        }
      >
        Button Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`BaseButton tests SecondaryWithIcon snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#49C4E9",
          "borderRadius": 7,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <test-file-stub
        fill="#FFFFFF"
        height={24}
        width={24}
      />
      <Text
        color="#FFFFFF"
        fontFamily="Rubik-Bold"
        fontSize={16}
        hasIcon={true}
        lineHeight={20}
        numberOfLines={1}
        size="Regular"
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "color": "#FFFFFF",
              "marginLeft": 8,
            },
          ]
        }
      >
        Button Text
      </Text>
    </View>
  </View>
</View>
`;
